### PR TITLE
fix: use canonical path in windows

### LINKED ISSUE #953: [Bug] `NvimTreeFindFile` do not reveal file in tree
**OS**
Win10 19043

**Neovim version**
NVIM v0.6.1
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compiled by runneradmin@fv-az152-430
Features: -acl +iconv +tui

**Nvim-tree version**
f938bcb 2022-02-08  (HEAD -> master, origin/master, origin/HEAD) fix: cleanup all dirs when hijacking window [kiyan]

**Describe the bug**
In windows, after open any file which attached by any LS configed by `nvim-lspconfig`, `NvimTreeFindFile` do not work properly on any further opened files attached by same LS. That is I can open nvim-tree but can not reveal the file in nvim-tree.

**To Reproduce**
1. Open a file in workspace and wait a few second until LS that attaches the file completes initialization
2. Open other file attached by the same LS. Note that no matter how you open it (by `e filename` or by `telescope` or by `goto definetion` of `LS`), this file's bufname is a full path starting with lower case of driver letter which cause the problem
3. `NvimTreeFindFile` or `NvimTreeOpen` can't put the cursor in that file in nvim-tree.

**Expected behavior**
`NvimTreeFindFile` or `NvimTreeOpen` can reveal the file in nvim-tree.

### PR DESCRIPTION
**Pull request by younger-1**
This fixes #953.

Relative upstream issue: https://github.com/neovim/neovim/issues/16331

### CODE REVIEW

File: lua/nvim-tree.lua [hunk 1]
```diff
@@ -188,6 +188,7 @@ function M.find_file(with_open)
   local bufname = vim.fn.bufname()
   local bufnr = api.nvim_get_current_buf()
   local filepath = vim.fn.fnamemodify(bufname, ':p')
+  filepath = utils.canonical_path(filepath)
```
Comment (kyazdani42) [inline lua/nvim-tree.lua:191]: could you avoid reassign ? 
```suggestion
  local filepath = utils.canonical_path(vim.fn.fnamemodify(bufname, ':p'))
```

File: lua/nvim-tree/utils.lua [hunk 2]
```diff
@@ -228,4 +230,13 @@ function M.path_normalize(path)
   return M.path_join(M.take(#components - num_dots * 2, components))
 end
 
+--- @param path string
+--- @return string
+function M.canonical_path(path)
+  if M.is_windows and path:match '^%a:' then
+    path = path:sub(1, 1):upper() .. path:sub(2)
```
Comment (kyazdani42) [inline lua/nvim-tree/utils.lua:237]: just use return here :) avoid reassignment please

**Comment by kyazdani42** [comment]
could you rebase :) ?

**Comment by kyazdani42** [comment]
Also i'm not sure for your issue with realpath, might be related to windows somehow ?

**Comment by younger-1** [comment]
> could you rebase :) ?

Done

**Review by kyazdani42 — APPROVED** [review]

**Comment by kyazdani42** [comment]
thanks :)

**Pull request merged**